crates/sui-core/src/authority_active/gossip/mod.rs

```diff
@@ -45,7 +45,7 @@ const REFRESH_FOLLOWER_PERIOD_SECS: u64 = 60;
 
 use super::ActiveAuthority;
 
-pub async fn gossip_process<A>(active_authority: &ActiveAuthority<A>, degree: usize)
+pub async fn gossip_process<A>(active_authority: &ActiveAuthority<A>, degree: usize) -> SuiResult
```

Comment:
The gossip process will be running as an independent component, so all the errors that are found should be handled within the component.

Reply:
But what if the component fails to start? Shouldn't that error be reportable to the caller?

Reply:
One of my next tasks is to instantiate this component via the resilient component library, which will provide the parental supervision which handles that.

Reply:
See https://github.com/MystenLabs/mysten-infra/tree/main/crates/component for more details

Reply:
@mystenmark I agree with you, i have a number of concerns that i've tried to raise about this as well